apis/Google.Storage.V1/Google.Storage.V1/SignedUrlUtils.cs

```diff
@@ -0,0 +1,396 @@
+        /// <param name="credentialFilePath">The path to the JSON key file for a service account. Must not be null.</param>
+        /// <param name="expiration">The point in time after which the Signed URL will be invalid. May be null, in which
+        /// case the Signed URL never expires.</param>
+        /// <param name="requestMethod">The HTTP request method for which the Signed URL is allowed to be used. May be null,
+        /// in which case GET will be used.</param>
+        /// <param name="requestHeaders">The headers which will be included with the request. May be null.</param>
+        /// <param name="contentHeaders">The headers for the content which will be included with the request.
+        /// May be null.</param>
+        /// <remarks>
+        /// <para>
+        /// When either of the headers collections are specified, there are certain headers which will be included in the
+        /// Signed URL's signature, and therefore must be included in requests made with the created URL. These are the
+        /// Content-MD5 and Content-Type content headers as well as any content or request header with a name starting
+        /// with "x-goog-". However, this does not apply to x-goog-encryption-key and x-goog-encryption-key-sha256 headers,
+        /// but they must be included with the request anyway if the desired content is encrypted with customer-supplied
+        /// encryption keys.
+        /// </para>
+        /// <para>
+        /// Note that when GET is specified as the <paramref name="requestMethod"/> (or it is null, in which case GET is
+        /// used), both GET and HEAD requests can be made with the created Signed URL.
+        /// </para>
+        /// <para>
+        /// See https://cloud.google.com/storage/docs/access-control/signed-urls for more information on Signed URLs.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">
+        /// The <paramref name="credentialFilePath"/> does not refer to a valid JSON service account key file.
+        /// </exception>
+        /// <returns>
+        /// The Signed URL which can be used to provide access to a bucket or object for a limited amount of time.
+        /// </returns>
+        public static string Create(
+            string bucket,
+            string objectName,
+            string credentialFilePath,
+            DateTimeOffset? expiration,
+            HttpMethod requestMethod = null,
+            Dictionary<string, IEnumerable<string>> requestHeaders = null,
+            Dictionary<string, IEnumerable<string>> contentHeaders = null)
+        {
+            StorageClientImpl.ValidateBucketName(bucket);
+            GaxPreconditions.CheckNotNull(credentialFilePath, nameof(credentialFilePath));
+
+            var credentials = ParseCredentials(credentialFilePath);
+            var timestamp = ((int?)((expiration - UnixEpoch)?.TotalSeconds))?.ToString();
```

Comment:
This is a duration rather than a timestamp, right? I'd also use specify CultureInfo.InvariantCulture when converting to a string.